### PR TITLE
Ritam add style

### LINKED ISSUE #6: Add Style
#### As a user I want to see a visually pleasing application.

__Story__:

Our web application is really starting taking shape. We have added a few
features that provide value to our customers, but some of our customers are
complaining that the website is a quite bland. It is time to spice it up by
applying a visually appealing, and easy to maintain user interface to our web
application.

__Acceptance criteria__:
- [x] The application has a header with navigation
- [x] Images are displayed in an aesthetically pleasing list of cards
- [x] Form errors are shown inline with the related input field
- [x] I know how to use Bootstrap components like buttons and navbars in my
  application.
- [ ] I know the very basics of CSS and SCSS

__Reference material__:
- http://v4-alpha.getbootstrap.com/getting-started/introduction/
- https://github.com/plataformatec/simple_form

__Dependencies__:
- Image Index

### PR DESCRIPTION
**Pull request by RitamBanerjee**
Closes #6

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [x] I have replaced `X` above with the associated issue number.
- [x] I have changed the issue's label to "pull request".
- [x] I have assigned the designated reviewer to this PR.
- [x] I have set the appropriate label on this PR.
- [x] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

File: app/assets/stylesheets/_base.scss [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
```
Comment (aaronpanch) [inline app/assets/stylesheets/_base.scss]: Ritam, I already like the way you've organized this! Did you come up with this yourself, or model it after someone? (p.s. there's an extra line here ;)

File: app/assets/stylesheets/_base.scss [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+a {
```
Comment (aaronpanch) [inline app/assets/stylesheets/_base.scss]: Is there a bootstrap variable you can override for this style?

File: app/assets/stylesheets/_base.scss [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  color: inherit;
+  text-decoration: none;
+}
+
+body {
+  background-color: gray;
```
Comment (aaronpanch) [inline app/assets/stylesheets/_base.scss]: I'm pretty sure theres a bootstrap variable for this too.

File: app/assets/stylesheets/images.scss [hunk 4]
```diff
@@ -1,3 +1,17 @@
-img{
-  max-width: 400px;
+.pic{
```
Comment (aaronpanch) [inline app/assets/stylesheets/images.scss]: I know this may be a bit picky, but I like a space after the class name:
```css
.pic {
}
```

File: app/controllers/images_controller.rb [hunk 5]
```diff
@@ -1,13 +1,14 @@
 class ImagesController < ApplicationController
   def new
+    @image = Image.new
   end
 
   def create
     @image = Image.new(permitted_image_params)
 
-    if !@image.save
+    if @image.invalid?
```
Comment (aaronpanch) [inline app/controllers/images_controller.rb]: Why did you change this from `.save`?  `.save` returns a boolean indicating whether or not it saved, which also runs all the validations.

File: app/views/images/_image_box.html.erb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<div class="d-flex-wrap p-1 pic_box">
+  <a href = <%= image_url(image.id)%>><%= image.title %>
+  <br>
```
Comment (aaronpanch) [inline app/views/images/_image_box.html.erb]: Ah, it's nice to avoid line breaks.  If you want these two elements on different lines, wrap one (or both) in a `p`.

File: app/views/images/_image_box.html.erb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+<div class="d-flex-wrap p-1 pic_box">
+  <a href = <%= image_url(image.id)%>><%= image.title %>
```
Comment (aaronpanch) [inline app/views/images/_image_box.html.erb]: There's a rails helper to generate links: `link_to`. If you want to add more than just a string of content, pass it a block.

File: app/views/images/index.html.erb [hunk 8]
```diff
@@ -1,26 +1,23 @@
...
+  </button>
+  <a class="navbar-brand logo" href= <%= images_url %>>ImGuru</a>
+  <div class="collapse navbar-collapse" id="navbarNavAltMarkup">
+    <div class="navbar-nav">
+      <a class="nav-item nav-link active" href= <%= images_url %>>Index<span class="sr-only">(current)</span></a>
+      <a class="nav-item nav-link" href= <%= new_image_url %>>New Image</a>
+    </div>
+  </div>
+</nav>
+<div class = 'aligner'>
```
Comment (aaronpanch) [inline app/views/images/index.html.erb]: Avoid putting spaces around the equals sign `=` here.

File: app/views/images/new.html.erb [hunk 9]
```diff
@@ -1,15 +1,21 @@
...
+    <span class="navbar-toggler-icon"></span>
+  </button>
+  <a class="navbar-brand" href=<%= images_url %>>ImGuru</a>
+  <div class="collapse navbar-collapse" id="navbarNavAltMarkup">
+    <div class="navbar-nav">
+      <a class="nav-item nav-link active" href= <%= images_url %>>Index<span class="sr-only">(current)</span></a>
+      <a class="nav-item nav-link" href= <%= new_image_url %>>New Image</a>
+    </div>
+  </div>
+</nav>
```
Comment (aaronpanch) [inline app/views/images/new.html.erb]: This looks like it's the same in both the index file and here.  If it's supposed to be on every page in the app, move it to `layouts/application.html.erb`

File: app/views/images/new.html.erb [hunk 10]
```diff
@@ -1,15 +1,21 @@
...
+
+<div class="Aligner pic_box inputtext">
+<%= simple_form_for @image, url: images_path do |f| %>
```
Comment (aaronpanch) [inline app/views/images/new.html.erb]: The indentation is off for this element.

Comment (aaronpanch) [reply app/views/images/new.html.erb]: I'm not sure you have to specify the `url` attribute here.  But I'm not 100% sure

File: app/views/images/index.html.erb [hunk 11]
```diff
@@ -1,26 +1,23 @@
...
+  <div class="flex-wrap aligner">
+  <% if @images.size > 0 %>
```
Comment (aaronpanch) [inline app/views/images/index.html.erb]: A more "rails way" of checking this, is to ask `@images.present?`

File: app/views/images/index.html.erb [hunk 12]
```diff
@@ -1,26 +1,23 @@
...
+      <% @images.each do |image| %>
+          <%= render partial: 'images/image_box', locals: {image: image} %>
```
Comment (aaronpanch) [inline app/views/images/index.html.erb]: Indentation

File: app/views/images/new.html.erb [hunk 13]
```diff
@@ -1,15 +1,21 @@
...
-      <%= f.submit %>
-    </p>
+<nav class="navbar navbar-toggleable-md navbar-inverse bg-inverse">
+  <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#navbarNavAltMarkup" aria-controls="navbarNavAltMarkup" aria-expanded="false" aria-label="Toggle navigation">
+    <span class="navbar-toggler-icon"></span>
+  </button>
+  <a class="navbar-brand" href=<%= images_url %>>ImGuru</a>
+  <div class="collapse navbar-collapse" id="navbarNavAltMarkup">
+    <div class="navbar-nav">
+      <a class="nav-item nav-link active" href= <%= images_url %>>Index<span class="sr-only">(current)</span></a>
```
Comment (aaronpanch) [inline app/views/images/new.html.erb]: You can also use the `link_to` helper here.

File: app/assets/stylesheets/_base.scss (hunk 1 above)
Comment (RitamBanerjee) [reply app/assets/stylesheets/_base.scss]: Haha I got a little inspiration from Joel
